npk/src/npk.rs

```diff
@@ -152,15 +154,15 @@ pub struct Npk<R: io::Read + io::Seek> {
 }
 
 impl<R: io::Read + io::Seek> Npk<R> {
```

Comment:
This is key: The signature requires `std` types but is async. Thats not ok from API perspective.
The `zip` crates needs a `std::io::Read`. We want to provide an api with `tokio::io::*`. Two things must be done:
When creating the `ZipArchive` the `tokio` types must be wrapped or converted into std types *and* every call to `ArchiveReader` must be wrapped in a `block_in_place` or `spawn_blocking` call.

Reply:
I am not sure how this can be done:
> When creating the ZipArchive the tokio types must be wrapped or converted into std types
Wouldn't this require some kind of adapter that wraps the async object and implements the synchronous `Read` and `Seek` traits?
```struct AsyncReadAdapter<R: AsyncRead + AsyncSeek> {
    inner: R,
}

impl<R> io::Read for AsyncReadAdapter<R>
where
    R: AsyncRead + AsyncSeek,
{
    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
        unimplemented!()
    }
}

impl<R> io::Seek for AsyncReadAdapter<R>
where
    R: AsyncRead + AsyncSeek,
{
    fn seek(&mut self, pos: SeekFrom) -> io::Result<u64> {
        unimplemented!()
    }
}```
If so, the required `read()` and `seek()` methods would be synchronous themselves but have to call async code at the same time.